### PR TITLE
Implemented and tested WorkCoverageRecord.bulk_add.

### PR DESCRIPTION
**Pull request by leonardr**
This branch implements a new method which takes a list of Works and makes sure every work has an up-to-date WorkCoverageRecord, whether that means inserting a new row into the database or updating an old row.

### CODE REVIEW

File: tests/test_model.py [hunk 1]
```diff
@@ -5727,6 +5727,64 @@ def test_add_for(self):
...
+            not_affected, irrelevant_operation, 
+            status=WorkCoverageRecord.SUCCESS
+        )
+
+        # Tell bulk_add to update or create WorkCoverageRecords for
+        # not_already_covered and already_covered, but not not_affected.
+        new_timestamp = datetime.datetime.utcnow()
+        new_status = WorkCoverageRecord.REGISTERED
+        WorkCoverageRecord.bulk_add(
+            [not_already_covered],
```
Comment (aslagle) [inline tests/test_model.py]: how did already_covered get updated when it's not in here?

Comment (leonardr) [reply tests/test_model.py]: Uh-oh.

**Comment by leonardr** [comment]
I made the test a lot more thorough and fixed some bugs.

File: model.py [hunk 2]
```diff
@@ -1497,6 +1498,70 @@ def add_for(self, work, operation, timestamp=None,
...
+        _db.execute(insert)
+
+        # Make sure that works that previously had a
+        # WorkCoverageRecord for this operation have their timestamp
+        # and status updated.
+        update = WorkCoverageRecord.__table__.update().where(
+            and_(WorkCoverageRecord.work_id.in_(work_ids),
+                 WorkCoverageRecord.operation==operation)
+        ).values(dict(timestamp=timestamp, status=status, exception=exception))
+        _db.execute(update)
```
Comment (aslagle) [inline model.py]: Does doing the insert before the update mean that the new records are updated unnecessarily? Maybe it doesn't make a difference, but doing the update first could be faster.

**Comment by leonardr** [comment]
I agree it doesn't make a big difference; I was thinking about the case where two processes run this method simultaneously and the SQL calls are interleaved. But even then I guess it doesn't matter which timestamp the records end up with.

**Pull request merged**